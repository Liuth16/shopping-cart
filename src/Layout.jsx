import { Outlet, Link } from "react-router-dom";
import styles from "./styles/layout.module.css";

const Layout = () => {
  return (
    <>
      <header>
        <nav>
          <ul className={styles["testList"]}>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/products">Products</Link>
            </li>
            <li>
              <Link to="/cart">Cart</Link>
            </li>
          </ul>
        </nav>
      </header>
      <main>
        <Outlet /> {/* This renders the matched child route */}
      </main>
      <footer>
        <div className="footerLinksContainer">
          <a href="#">Link 1</a>
          <a href="#">Link 1</a>
          <a href="#">Link 1</a>
          <a href="#">Link 1</a>
        </div>
        <div className="footerSocialsContainer">
          <a href="">
            <img src="../src/assets/twitter.svg" alt="" />
          </a>
          <a href="">
            <img src="../src/assets/instagram.svg" alt="" />
          </a>
          <a href="">
            <img src="../src/assets/facebook.svg" alt="" />
          </a>
        </div>
        <p>Â© 2025 My Store</p>
      </footer>
    </>
  );
};

export default Layout;
