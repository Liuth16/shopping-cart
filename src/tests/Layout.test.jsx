import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import Layout from "../Layout";

// Dummy component for nested routes
function DummyPage({ text }) {
  return <div>{text}</div>;
}

describe("Layout", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders navigation links", () => {
    render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );
    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /products/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /cart/i })).toBeInTheDocument();
  });

  it("renders footer links and social icons", () => {
    render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );
    expect(screen.getByText(/about us/i)).toBeInTheDocument();
    expect(screen.getByText(/contact/i)).toBeInTheDocument();
    expect(screen.getByText(/shipping policy/i)).toBeInTheDocument();
    expect(screen.getByText(/returns & refunds/i)).toBeInTheDocument();
    expect(screen.getByAltText(/twitter/i)).toBeInTheDocument();
    expect(screen.getByAltText(/instagram/i)).toBeInTheDocument();
    expect(screen.getByAltText(/facebook/i)).toBeInTheDocument();
  });

  it("renders children via Outlet", () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route index element={<DummyPage text="Nested Content" />} />
          </Route>
        </Routes>
      </MemoryRouter>
    );
    expect(screen.getByText(/nested content/i)).toBeInTheDocument();
  });

  it("navigates to Home, Products, and Cart when links are clicked", async () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route index element={<DummyPage text="Home Page" />} />
            <Route path="products" element={<DummyPage text="Products Page" />} />
            <Route path="cart" element={<DummyPage text="Cart Page" />} />
          </Route>
        </Routes>
      </MemoryRouter>
    );

    // Home link
    await userEvent.click(screen.getByRole("link", { name: /home/i }));
    expect(screen.getByText(/home page/i)).toBeInTheDocument();

    // Products link
    await userEvent.click(screen.getByRole("link", { name: /products/i }));
    expect(screen.getByText(/products page/i)).toBeInTheDocument();

    // Cart link
    await userEvent.click(screen.getByRole("link", { name: /cart/i }));
    expect(screen.getByText(/cart page/i)).toBeInTheDocument();
  });
});