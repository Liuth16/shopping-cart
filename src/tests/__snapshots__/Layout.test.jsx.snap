// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > matches snapshot 1`] = `
<div>
  <header
    class="_header_f3cea2"
  >
    <nav
      class="_nav_f3cea2"
    >
      <ul
        class="_navList_f3cea2"
      >
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/products"
          >
            Products
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/cart"
          >
            Cart
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="_main_f3cea2"
  />
  <footer
    class="_footer_f3cea2"
  >
    <div
      class="_footerContent_f3cea2"
    >
      <div
        class="_footerLinksContainer_f3cea2"
      >
        <a
          href="#"
        >
          About Us
        </a>
        <a
          href="#"
        >
          Contact
        </a>
        <a
          href="#"
        >
          Shipping Policy
        </a>
        <a
          href="#"
        >
          Returns & Refunds
        </a>
      </div>
      <div
        class="_footerSocialsContainer_f3cea2"
      >
        <a
          href="#"
        >
          <img
            alt="Twitter"
            src="/twitter.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Instagram"
            src="/instagram.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Facebook"
            src="/facebook.svg"
          />
        </a>
      </div>
    </div>
    <p
      class="_copyright_f3cea2"
    >
      © 2025 My Store
    </p>
  </footer>
</div>
`;

exports[`Layout > renders headline 1`] = `
<div>
  <header
    class="_header_f3cea2"
  >
    <nav
      class="_nav_f3cea2"
    >
      <ul
        class="_navList_f3cea2"
      >
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/products"
          >
            Products
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/cart"
          >
            Cart
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="_main_f3cea2"
  />
  <footer
    class="_footer_f3cea2"
  >
    <div
      class="_footerContent_f3cea2"
    >
      <div
        class="_footerLinksContainer_f3cea2"
      >
        <a
          href="#"
        >
          About Us
        </a>
        <a
          href="#"
        >
          Contact
        </a>
        <a
          href="#"
        >
          Shipping Policy
        </a>
        <a
          href="#"
        >
          Returns & Refunds
        </a>
      </div>
      <div
        class="_footerSocialsContainer_f3cea2"
      >
        <a
          href="#"
        >
          <img
            alt="Twitter"
            src="/twitter.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Instagram"
            src="/instagram.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Facebook"
            src="/facebook.svg"
          />
        </a>
      </div>
    </div>
    <p
      class="_copyright_f3cea2"
    >
      © 2025 My Store
    </p>
  </footer>
</div>
`;

exports[`Layout > renders layout page 1`] = `
<div>
  <header
    class="_header_f3cea2"
  >
    <nav
      class="_nav_f3cea2"
    >
      <ul
        class="_navList_f3cea2"
      >
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/products"
          >
            Products
          </a>
        </li>
        <li>
          <a
            class="_navLink_f3cea2"
            data-discover="true"
            href="/cart"
          >
            Cart
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="_main_f3cea2"
  />
  <footer
    class="_footer_f3cea2"
  >
    <div
      class="_footerContent_f3cea2"
    >
      <div
        class="_footerLinksContainer_f3cea2"
      >
        <a
          href="#"
        >
          About Us
        </a>
        <a
          href="#"
        >
          Contact
        </a>
        <a
          href="#"
        >
          Shipping Policy
        </a>
        <a
          href="#"
        >
          Returns & Refunds
        </a>
      </div>
      <div
        class="_footerSocialsContainer_f3cea2"
      >
        <a
          href="#"
        >
          <img
            alt="Twitter"
            src="/twitter.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Instagram"
            src="/instagram.svg"
          />
        </a>
        <a
          href="#"
        >
          <img
            alt="Facebook"
            src="/facebook.svg"
          />
        </a>
      </div>
    </div>
    <p
      class="_copyright_f3cea2"
    >
      © 2025 My Store
    </p>
  </footer>
</div>
`;
